wom/src/main/scala/wom/graph/GraphNode.scala

```diff
@@ -11,6 +11,10 @@ trait GraphNode {
 
   def name: String
 
+  final override def equals(other: Any): Boolean = super.equals(other)
+
+  final override def hashCode: Int = super.hashCode()
```

Comment:
This is fine but ToL, we could be more explicit about what we want here, eg
```
final override def equals(other: Any): Boolean = this eq other
final override def hashCode: Int = System.identityHashCode(this)
```

Reply:
Though I think @kshakir has pointed out in the past that `System.identityHashCode` is not necessarily the proxy-to-object-identity that developers like to think it is...

Reply:
this.eq and System.identityHashCode only work with AnyRef, not Any, argument, so above suggestion doesn't work. Should we add code to treat AnyVal and AnyRef differently, or leave it as it is?